lib/spack/spack/cmd/install.py

```diff
@@ -84,6 +85,9 @@ def setup_parser(subparser):
         default=None,
         help="phase to stop after when installing (default None)",
     )
+    subparser.add_argument(
+        "-p", "--nprocs", type=int, default=4, help="number of concurrent packages to build"
```

Comment:
Yeah both of those work for me.

Reply:
I don't think either of those effectively convey to the user what the flag does without them needing to read the documentation. I'd prefer nprocs to either of those, regardless of the potential confusion with spack processes vs system processes.

Reply:
I like concurrent being part of the name as well, it adds clarity. I don't have much of a preference between --concurrent-builds or --concurrent-packages, does anyone else?

Reply:
--concurrent-builds seems to be the most descriptive but I think it might be less so once the job server implementation gets done. 

As I think about this more I have questions on how this flag and the relationship with -j will evolve with the growth of this feature. I propose we target flags that will encapsulate the end product intention instead of the phase 1 implementation. 

Right now we are hard coding lanes of parallelism/concurrency. Presumably with a jobserver it will dynamically adapt based on the available resources and lanes of parallelism open in the graph. Here are N tokens (processors) and if you have N concurrent builds avail you could launch them all at once and throttle down concurrency as but increase resources in each instance as the graph narrows. 

If we have dynamic resources in terms of concurrent builds would this flag then be a cap on lanes of parallelism open (I could see value in this to avoid slow serial builds), or just like the total number of tokens available and -j be the max cores for a particular build?

Reply:
Okay that makes sense to me. I could also see the case where I don't want to allow parallelism to drop below a certain level so I want to limit concurrency. So then I would propose the flag be `--max-concurrency` to express it is an upper bound.

Reply:
I've decided to go with `-p/--concurrent-packages`. Thank you for your suggestions and ensuring that we picked an adequate and representative name :)